adapters/avail/src/spec/address.rs

```diff
@@ -0,0 +1,48 @@
+use core::fmt::{Display, Formatter};
+use std::hash::Hash;
+use std::str::FromStr;
+
+use primitive_types::H256;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Eq, Hash)]
+pub struct AvailAddress(pub [u8; 32]);
```

Comment:
Nit: Is there strong reason to publicly expose internal bytes array?

Reply:
Fixed by #980